src/plugins/irc-events/join.js

```diff
@@ -15,12 +15,12 @@ module.exports = function(irc, network) {
 				name: data.channel,
 				state: Chan.State.JOINED,
 			});
-			network.channels.push(chan);
-			client.save();
+
 			client.emit("join", {
 				network: network.id,
-				chan: chan.getFilteredClone(true),
+				index: network.addChannel(chan),
 			});
+			client.save();
```

Comment:
Any consequence in switching `emit`/`save` and, maybe more importantly, not using `chan.getFilteredClone` anymore?
I'm probably missing something but I'm not sure why we don't send `chan` property anymore.

Reply:
> Any consequence in switching emit/save

There's only 1 that I can think, and that's that if the server crashes between the 2 commands, then the channel won't be saved in the user's file. I can extract network.addChannel out from the emit itself if you prefer.

> not using chan.getFilteredClone anymore?
> I'm probably missing something but I'm not sure why we don't send chan property anymore.

Not missing anything, refactor error. I didn't test this 1 case after refactoring to move the index calculation into the network. My bad.